src/logger.rs

```diff
@@ -0,0 +1,113 @@
+pub(crate) use lightning::util::logger::Logger;
+use lightning::util::logger::Record;
+use lightning::util::ser::Writer;
+
+use chrono::Utc;
+
+use std::fs;
+
+pub(crate) struct FilesystemLogger {
+	data_dir: String,
+}
+
+impl FilesystemLogger {
+	pub(crate) fn new(data_dir: String) -> Self {
```

Comment:
logger should take logs_file_path/logfilename as arg instead of data_dir.
that way, i could use this to log to different file if needed in other parts of code.

Reply:
Done with 28b9adc12dca45be1fdb119c968a9dedbd003ec8.